### DISCUSSION TITLE
Errors: no manifest attribute in jar for Mac, and skip non existing resourceDirectory **Error:, when directory does exist?

### DISCUSSION THREAD

**Discussion by Sushi-roll89** [apache/netbeans#7520] [General]
### Apache NetBeans version

Apache NetBeans 22

### What happened

I can not open jar files, I keep getting these errors even though the directory does exist.

### Language / Project Type / NetBeans Component

Java maven

### How to reproduce

n/a

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Mac Monterey and windows 11

### JDK

jdk 21, 17, and 22

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

no

### Are you willing to submit a pull request?

Yes

**Comment by matthiasblaesing** [comment]
What do you want to do? What is the error you are seeing? You literally don't give anything to help you.

**Reply by Sushi-roll89** [reply]
I Apologize for my late response, I am a student, and work full time. You see I would have given you the path but it has my name (personal information). I figured someone who has worked with java may at some point run through the same problem; Many YouTube videos on this subject. Let me better explain, when I attempt to create a JAR file, the directory of the project does not get recognized after the clean and build. What it does is it deletes the so called "none existing directory. Then, it creates it in the same directory that claims not exist. So, I tried a few things like creating it with, Maven, Gradle, and Ant, they all provided different options, however no progress was made. My issue is I have gotten a few errors which I tried to work out, but no matter what I can't create a JAR file (which I heard can be converted to an exe. file).  However, I must successfully create a JAR first. 

I have attempted the following:

1. Adding the Main-Class: in .MF doc.
2. editing the project.properties file to run the path manually.
3. building with dependencies
4. editing the Pom.xml file, including the build.xml file. 
5. Opening with zip file to edit the Main-Class: 
6. creating a run.bat file to launch it. 
7. Used mkdir for the path that claimed to be missing. 
8. Tried many things on the terminal. 

All this was mostly on maven. My guess is that if I wanted this jar to work it should have a path that every user can access? This guess comes from a time I was installing a windows core server 11 os system to a  server. I had the program in what I thought was the root directory until I found out that it worked only work if it was outside of all directories, in other words within no folders (I know this guess is pretty out of left field but worth noting).

These are a few of the things I tried and none of them bared any results. Could it be that my computer is from 2015?
or lacking plug-ins, version types, or IDEs? I'm taking COSC 1437, although this may not be part of my assignments, it has been mentioned in two separate chapters, I figured it should be something I should be capable of doing as a newbie programmer. I am new to the GitHub. The program is a simple console conversion program, also, after some time, I was able to get the terminal to run it, but it came with an error, yet still found the path and could play the conversion program whose path does not exist according to the "create and build" console messages.

**Reply by matthiasblaesing** [reply]
Have a look here: 

- https://medium.com/@AlexanderObregon/setting-up-a-maven-project-in-netbeans-a-step-by-step-guide-e1089da7ac
- https://maven.apache.org/guides/getting-started/maven-in-five-minutes.html
- https://www.baeldung.com/executable-jar-with-maven

**Reply by Sushi-roll89** [reply]
Everything this guy said is default stuff. My problem is passed this.

**Reply by matthiasblaesing** [reply]
I read to many attempts in your very short description. You randomly mix build systems and don't form a description of what you did, what you saw, what you expected. I suggest StackOverflow for general programming questions or questions how to use your IDE.

**Reply by Sushi-roll89** [reply]
Yes, I had to try so many different options with different builds, and still kept getting the same problems, I watched dozens of videos with so called solutions but none of them worked. I'll provide more information. I'm just getting ready to work on this stuff in a few moments so I'll update more with extreme detail.

**Reply by matthiasblaesing** [reply]
STOP: Don't run around. Keep focused. Don't switch build systems. Do a minimal demonstrator. Before everybody created $random video for $trivial problem, people created simple, self contained samples with description what they did (with steps), what they expected and what they saw. Doing this nearly always solved the issue, because with this approach you analyse the problem and find your own failures.

**Reply by Sushi-roll89** [reply]
I agree with you, when I switching builds it was after 12 hours of trouble
shooting and getting no where on maven. Then I switch build and went back
to maven. I did terminal work with maven, and more. I even tried on Windows
and Mac. I figured someone had already run into this problem.
